packages/devtools/src/sourceMap/loadOriginalSourceLoc.ts

```diff
@@ -0,0 +1,178 @@
+// when more than one source maps appear in a source file: `${source}:${line}:${column}` -> source map JSON
+const sourceMapJSONs: Map<string, RawSourceMap> = new Map();
+
+const getComputedSouceMapJSON = ({ source, line, column }: MappedPosition) => {
+  return sourceMapJSONs.get(source) ?? sourceMapJSONs.get(`${source}:${line}:${column}`);
+};
+
+export async function loadOriginalSourceLoc(sourceUrlWithLoc: string): Promise<MappedPosition> {
+  const paths = sourceUrlWithLoc.split(':');
+  const column = Number(paths.pop());
+  const line = Number(paths.pop());
+  const source = paths.join(':');
+  if (Number.isNaN(line) || Number.isNaN(column)) {
+    return { source, line: 1, column: 0 };
+  }
+  const sourceLoc: MappedPosition = { source, line, column };
+
+  const sourceMapJSON = getComputedSouceMapJSON(sourceLoc);
+  if (sourceMapJSON) {
+    return getOriginalPosition(sourceMapJSON, sourceLoc);
+  }
+
+  try {
+    return extractAndLoadSourceMapJSON(sourceLoc);
+  } catch (error) {
+    console.warn(`[Griffel devtools] unable to load source map for ${sourceUrlWithLoc}`);
+    console.warn(error);
+    return sourceLoc;
+  }
+}
+
+// inspired by https://github.com/facebook/react/blob/b66936ece7d9ad41a33e077933c9af0bda8bff87/packages/react-devtools-shared/src/hooks/parseHookNames/loadSourceAndMetadata.js#L135
+async function extractAndLoadSourceMapJSON(sourceLoc: MappedPosition): Promise<MappedPosition> {
+  const { source, line, column } = sourceLoc;
+
+  const runtimeSourceCode = await fetchFiles(source);
+
+  const externalSourceMapURLs = [];
+  let sourceMapNum = 0;
+  const sourceMapRegex = / ?sourceMappingURL=([^\s'"]+)/gm;
+
+  let sourceMappingURLMatch = sourceMapRegex.exec(runtimeSourceCode);
+  if (!sourceMappingURLMatch) {
+    throw new Error(`[Griffel devtools] extractAndLoadSourceMapJSON() no source map found in ${source}`);
+  }
+
+  while (sourceMappingURLMatch) {
+    sourceMapNum++;
+    const sourceMappingURL = sourceMappingURLMatch[1];
+
+    const hasInlineSourceMap = sourceMappingURL.indexOf('base64,') >= 0;
+    if (hasInlineSourceMap) {
+      try {
+        // Web apps like Code Sandbox embed multiple inline source maps.
+        // In this case, we need to loop through and find the right one.
+        const trimmed = sourceMappingURL.match(/base64,([a-zA-Z0-9+/=]+)/)?.[1] ?? '';
+        const decoded = decodeBase64String(trimmed);
+        const sourceMapJSON = JSON.parse(decoded);
+
+        // TODO optionally turn on debug message
+        // if (__DEBUG__) {
+        //   console.groupCollapsed('[Griffel devtools] extractAndLoadSourceMapJSON() Inline source map');
+        //   console.log(sourceMapJSON);
+        //   console.groupEnd();
+        // }
+
+        if (sourceMapIncludesSource(sourceMapJSON, getFilePath(source))) {
+          // store computed sourceMapJSON
+          if (sourceMapNum > 1) {
+            sourceMapJSONs.set(`${source}:${line}:${column}`, sourceMapJSON);
```

Comment:
I'm interested to know why can't all sourcemaps be stored with line and column number ? what's the difference vs just source ?

Reply:
When there's one source map per source, only using `source` url as key is good enough.
When there's multiple source map per source, one `source` url will be related with multiple source maps. In this case I use `${source}:${line}:${column}` as key

Reply:
OK but is there any difference in usage ? this code can be simplified a lot if you stored all sourcemaps with the `source:line:column` format

Reply:
Yeah there is a difference:
Consider a file `http://myfile` which contains only one source map.

When the input is `http://myfile:1:2`, we do: read file -> find source map -> parse source map and get JSON-> locate line1 col2 in source map JSON -> done.

And the next time input could be `http://myfile:10:20` (another makeStyles hook in same file). If we store using `source:line:column`, we cannot reuse the source map JSON we get from the previous step. But if we store using just `source` as key, we can get source map JSON immediately.

Reply:
Ah got it thx